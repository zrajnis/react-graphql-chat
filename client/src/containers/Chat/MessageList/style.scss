.message-list {
  border: 1px solid $white;
  overflow-y: scroll;
  padding: 0 15px;
  width: 720px;
}
