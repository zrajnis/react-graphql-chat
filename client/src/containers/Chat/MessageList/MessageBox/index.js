import React from 'react'
import PropTypes from 'prop-types'

import './style.scss'

const MessageBox = ({ message, myMessage }) => (
  <li styleName={myMessage ? 'my-message-wrapper' : 'chat-message-wrapper'}>
    <div styleName='chat-message'>
      {!myMessage && <span>{message.from}:</span>}
      <p>
        {message.content}
      </p>
    </div>
  </li>
)

MessageBox.propTypes = {
  message: PropTypes.shape({
    content: PropTypes.string.isRequired,
    from: PropTypes.string.isRequired
  }).isRequired,
  myMessage: PropTypes.bool.isRequired
}

export default MessageBox
