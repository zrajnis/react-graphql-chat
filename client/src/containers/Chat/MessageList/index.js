import React from 'react'
import PropTypes from 'prop-types'

import MessageBox from 'containers/Chat/MessageList/MessageBox'
import './style.scss'

const MessageList = ({ allMessages, username }) =>
  <div styleName='message-container'>
    {allMessages.map(message => (
      <MessageBox
        key={message.id}
        message={message}
        myMessage={message.from === username}
      />
    ))}
  </div>

MessageList.propTypes = {
  allMessages: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.string.isRequired,
      from: PropTypes.string.isRequired
    })
  ),
  username: PropTypes.string.isRequired
}

export default MessageList
