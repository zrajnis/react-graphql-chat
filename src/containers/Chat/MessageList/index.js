import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import MessageBox from 'containers/Chat/MessageList/MessageBox'

const MessageList = ({ allMessages, from }) =>
  <Fragment>
    {allMessages.map(message => (
      <MessageBox
        key={message.id}
        message={message}
        myMessage={message.from === from}
      />
    ))}
  </Fragment>

MessageList.propTypes = {
  allMessages: PropTypes.arrayOf(
    PropTypes.shape({
      content: PropTypes.string.isRequired,
      from: PropTypes.string.isRequired
    })
  ),
  from: PropTypes.string.isRequired
}

export default MessageList
