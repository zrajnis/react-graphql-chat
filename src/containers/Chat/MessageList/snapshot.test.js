import React from 'react'
import renderer from 'react-test-renderer'
import MessageList from 'containers/Chat/MessageList'

describe('Message list component snapshot', () => {
  const from = 'testUser'
  const allMessages = [{
    content: 'test content 123',
    from,
    id: 1
  }, {
    content: 'test content 456',
    from: 'testUser1',
    id: 2
  }, {
    content: 'test content 678',
    from: 'testUser2',
    id: 3
  }]

  it('renders the message list', () => {
    const tree = renderer.create(
      <MessageList
        allMessages={allMessages}
        from={from}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
