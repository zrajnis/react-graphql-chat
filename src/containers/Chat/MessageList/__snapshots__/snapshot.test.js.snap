// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message list component snapshot renders the message list 1`] = `
<div
  className="message-container--28YesOzC"
>
  <div
    className="my-message-wrapper--h93b8ATj"
  >
    <div
      className="chat-message--QnC2_lx7"
    >
      <p>
        test content 123
      </p>
    </div>
  </div>
  <div
    className="chat-message-wrapper--hp6HY0Bp"
  >
    <div
      className="chat-message--QnC2_lx7"
    >
      <span>
        testUser1
        :
      </span>
      <p>
        test content 456
      </p>
    </div>
  </div>
  <div
    className="chat-message-wrapper--hp6HY0Bp"
  >
    <div
      className="chat-message--QnC2_lx7"
    >
      <span>
        testUser2
        :
      </span>
      <p>
        test content 678
      </p>
    </div>
  </div>
</div>
`;
